source/auth/auth.rst

```diff
@@ -232,7 +232,11 @@ MONGODB-X509
 
 :since: 2.6
 
-MONGODB-X509 is the usage of X-509 certificates to validate a client.  The server will use the distinguished subject name of the client certificate in the SSL negotiation to authenticate. The driver will be required to supply the distinguished subject name outside of the SSL negotiation to the server using the "authenticate" command.
+MONGODB-X509 is the usage of X-509 certificates to validate a client.  The server will use the distinguished subject name of the client certificate in the SSL negotiation to authenticate. The driver MAY supply the distinguished subject name outside of the SSL negotiation to the server using the "authenticate" command.
+
+:since: 3.4
+
+MongoDB 3.4 no longer requires the user or the drivers to provide the username. Drivers that have the ability to automatically extract the distinguished subject name from a certificate MAY do so when no username is otherwise provided for backwards compatibility of earlier MongoDB versions. A driver SHOULD NOT preemptively error when no username is provided.
```

Comment:
The problem I see is that at the time clients are starting up, checking configuration options, etc., they don't know what version of the server they will be talking to.  And once they talk to a 3.4 server, they later wind up talking to a downgraded server.  So the timing of when to tell the user they supplied.

Earlier, the spec says that username is always required.  And under errors, it says to report errors as early as possible.

I think this PR needs more thought and need to address *all* the spec holistically, not just a note in this section.